packages/create-llama/templates/types/streaming/express/src/controllers/llamaindex-stream.ts

```diff
@@ -4,27 +4,25 @@ import {
   trimStartOfStreamHelper,
   type AIStreamCallbacksAndOptions,
 } from "ai";
+import { Response } from "llamaindex";
 
-function createParser(res: AsyncGenerator<any>) {
+function createParser(res: AsyncIterable<Response>) {
   const trimStartOfStream = trimStartOfStreamHelper();
   return new ReadableStream<string>({
     async pull(controller): Promise<void> {
-      const { value, done } = await res.next();
-      if (done) {
-        controller.close();
-        return;
-      }
-
-      const text = trimStartOfStream(value ?? "");
-      if (text) {
-        controller.enqueue(text);
+      for await (const message of res) {
```

Comment:
@thucpn good catch! We're using here now an asynciterable. But better not consume all of it in the pull function (`pull` is called for pulling just one token).

You'll still have to use `next` by doing `let it = res[Symbol.asyncIterator]();`

see
https://github.com/vercel/ai/blob/b2349746892a292cfbbda5a59e8a125f53aa55b5/packages/core/streams/ai-stream.ts#L263